scripts/generate-test-fixtures/generate-hcx.ts

```
@@ -270,6 +272,22 @@ function generateInvalidHcx(errorType: ErrorType, baseNetwork?: any[]): any {
       }
       return cyclic
 
+    case 'not-dag-single-interaction':
+      // Same cycles as 'not-dag', but every edge shares one interaction type.
+      // The interaction heuristic therefore cannot detect this hierarchy, so it
+      // exercises the graceful failure of the circle packing builder itself
+      // (issue #630).
+      const uniformCyclic = generateInvalidHcx('not-dag', baseNetwork)
```

Biome (2.5.3)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

(lint/correctness/noSwitchDeclarations)